src/netif.c

```diff
@@ -719,27 +719,105 @@ static int set_ipv6_forwarding(const struct ip_setting_handler *handler, struct
-  { "ipv6_accept_ra", prep_ipv6_accept_ra, set_ipv6_accept_ra, get_ipv6_accept_ra, 0, 0 },
-  { "ipv6_accept_ra_pinfo", prep_ipv6_accept_ra_pinfo, set_ipv6_accept_ra_pinfo, get_ipv6_accept_ra_pinfo, 0, 0 },
-  { "ipv6_forwarding", prep_ipv6_forwarding, set_ipv6_forwarding, get_ipv6_forwarding, 0, 0 },
+  { .name = "ipv4_address",
+    .prep = prep_ipaddr_ioctl,
+    .set  = set_ipaddr_ioctl,
+    .get  = get_ipaddr_ioctl,
+    .ioctl_set = SIOCSIFADDR,
+    .ioctl_get = SIOCGIFADDR,
+  },
+  { .name = "ipv4_subnet_mask",
+    .prep = prep_ipaddr_ioctl,
+    .set  = set_ipaddr_ioctl,
+    .get  = get_ipaddr_ioctl,
+    .ioctl_set = SIOCSIFNETMASK,
+    .ioctl_get = SIOCGIFNETMASK,
+  },
+  { .name = "ipv4_broadcast",
+    .prep = prep_ipaddr_ioctl,
+    .set  = set_ipaddr_ioctl,
+    .get  = get_ipaddr_ioctl,
+    .ioctl_set  = SIOCSIFBRDADDR,
+    .ioctl_get  = SIOCGIFBRDADDR,
+  },
+  { .name = "ipv6_address",
+    .prep = prep_ipaddr6_ioctl,
+    .set  = set_ipaddr6_ioctl,
+    .get  = get_ipaddr6,
+    .ioctl_set = SIOCSIFADDR,
+    .ioctl_get = 0,
+  },
+  { .name = "ipv4_gateway",
+    .prep = prep_default_gateway,
+    .set  = set_default_gateway,
+    .get  = get_default_gateway,
+    .ioctl_set = 0,
+    .ioctl_get = 0,
+  },
+  { .name = "ipv6_gateway",
+    .prep = prep_default_gateway6,
+    .set  = set_default_gateway6,
+    .get  = get_default_gateway6,
+    .ioctl_set = 0,
+    .ioctl_get = 0,
+  },
+  { .name = "mac_address",
+    .prep = prep_mac_address_ioctl,
+    .set  = set_mac_address_ioctl,
+    .get  = get_mac_address_ioctl,
+    .ioctl_set = SIOCSIFHWADDR,
+    .ioctl_get = SIOCGIFHWADDR,
+  },
+  { .name = "ipv6_autoconf",
+    .prep = prep_ipv6_autoconf,
+    .set  = set_ipv6_autoconf,
+    .get  = get_ipv6_autoconf,
+    .ioctl_set = 0,
+    .ioctl_get = 0,
+  },
+  { .name = "ipv6_disable",
+    .prep = prep_ipv6_disable,
+    .set  = set_ipv6_disable,
+    .get  = get_ipv6_disable,
+    .ioctl_set = 0,
+    .ioctl_get = 0,
+  },
+  { .name = "ipv6_accept_ra",
+    .prep = prep_ipv6_accept_ra,
+    .set  = set_ipv6_accept_ra,
+    .get  = get_ipv6_accept_ra,
+    .ioctl_set = 0,
+    .ioctl_get = 0,
+  },
+  { .name = "ipv6_accept_ra_pinfo",
+    .prep = prep_ipv6_accept_ra_pinfo,
+    .set  = set_ipv6_accept_ra_pinfo,
+    .get  = get_ipv6_accept_ra_pinfo,
+    .ioctl_set = 0,
+    .ioctl_get = 0,
+  },
+  { .name = "ipv6_forwarding",
+    .prep = prep_ipv6_forwarding,
+    .set  = set_ipv6_forwarding,
+    .get  = get_ipv6_forwarding,
+    .ioctl_set = 0,
+    .ioctl_get = 0,
+  },
+  { .name = "-ipv6_address",
+    .prep = prep_ipaddr6_ioctl,
+    .set  = remove_ipaddr6_ioctl,
+    .get  = NULL,
+    .ioctl_set = SIOCDIFADDR,
+    .ioctl_get = 0,
+  },
+  { .name = NULL, } /* Setting-up a guard */
 };
-#define HANDLER_COUNT (sizeof(handlers) / sizeof(handlers[0]))
+#define HANDLER_COUNT ((sizeof(handlers)-1) / sizeof(handlers[0])) /* -1 is for the guard at the end of the array */
```

Comment:
I think we need to run this by Brian--this syntax loses the ability to compile with C++

Reply:
is there a reason we'd want it for C++?

Reply:
The nerves people selected the c99 standard as their basis. It includes the linux header files full of .something = some value. I daresay it would not compile with c++ in any way...
If it would. Furthermore
missing declarations of
#ifdef __cplusplus
extern "C" {
#endif

(...)
#ifdef __cplusplus
}
#endif

would make the linking stage failing for the c++..